pkg/cli/cli.go

```diff
@@ -351,11 +351,15 @@ func validatePlugins(plugins ...plugin.Base) error {
 					pluginName, projectVersion, err)
 			}
 		}
+
+		// fixme: the error `two plugins have the same name` is faced when we have
+		// more than on version of the same plugin with diff support abilities
+		// ----
 		// Check for duplicate plugin names. Names outside of a version can
 		// conflict because multiple project versions of a plugin may exist.
-		if _, seen := pluginNameSet[pluginName]; seen {
-			return fmt.Errorf("two plugins have the same name: %q", pluginName)
-		}
+		//if _, seen := pluginNameSet[pluginName]; seen {
```

Comment:
We need to fix plugin resolution before bumping plugin version.

Reply:
@estroz it is fixed. Also, we have tracked https://github.com/kubernetes-sigs/kubebuilder/issues/1526

Can we move forward here? Do you think that is required we should do a PR just to fix the check?